### PR TITLE
fix: error during persisted transaction approval

### LINKED ISSUE #20604: [Bug]: `Error during persisted transaction approval (Error#9)`
### Describe the bug

**Problem**: whenever I have a pending transaction, if I disable the wallet and enable it again, whenever I reject that transaction I get the error  `Error during persisted transaction approval (Error#9)`

Screenshot from 2023-08-25 14-58-50

https://github.com/MetaMask/metamask-extension/assets/54408225/58ff6acb-6ac1-4588-80b7-5df48ff440b1

### Steps to reproduce

1. Trigger any transaction
2. Disable the wallet from chrome://extensions/ (switch button)
3. Enable it again
4. Unlock MM
5. Reject the transaction
6. See error

### Error messages or log output

_No response_

### Version

10.34.5 but seems pre-existing issue

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation
This pull request addresses the issue that occurs when a pending transaction is present, the extension is disabled and then re-enabled, and subsequently, the user rejects the transaction. The goal is to prevent the logging of the error message "User rejected the request", while still capturing and logging any other types of errors.

Fixes https://github.com/MetaMask/metamask-extension/issues/20604

### After

[fix-error-during-persisted-transaction-approval.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/518c844a-4ae0-443c-a0bc-34591e3bc43f)

## Manual Testing Steps
- Trigger any transaction
- Disable the wallet from chrome://extensions/ (switch button)
- Enable it again
- Unlock MM
- Reject the transaction
- Check console

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by vinistevam** [comment]
> Pending transactions should be discarded when the extension is restarted (at least they used to be). Was that changed intentionally? If not, any idea why this is being persisted now?

From what I recall we added the initApprovals method in the extension as testing showed that pending transactions were persisting after restart so we needed to create the approval requests also. I've double-checked it with our team just to make sure the is no additional info around this subject. I also checked out the commit https://github.com/MetaMask/metamask-extension/commit/0b1354e44686ebeffc01dc5d870e051bc5e2b981 prior to our changes in the `TransactionController` and the transaction request remained once the extension is restarted, different from signatures for instance.

[Screencast from 29-08-2023 11:22:13.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/c6fa51d2-e699-45b0-95d4-3ddceb7bfef5)

**Review by matthewwalsh0 — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**
